File: QIQO.Monitor.Api/ViewModels/Monitor.cs
using QIQO.Monitor.Core.Contracts;
using QIQO.Monitor.SQLServer.Data;
using System.Collections.Generic;

namespace QIQO.Monitor.Api
{
    public class Monitor : IModel
    {
        public Monitor(MonitorData monitorData)
        {
            MonitorKey = monitorData.MonitorKey;
            MonitorName = monitorData.MonitorName;
            MonitorType = (MonitorType)monitorData.MonitorTypeKey;
            MonitorLevel = (MonitorLevel)monitorData.LevelKey;
            MonitorCategory = (MonitorCategories)monitorData.CategoryKey;
        }
        public Monitor(MonitorData monitorData, List<Query> queries) : this(monitorData) => Queries = queries;
        public Monitor(MonitorData monitorData, List<Query> queries, List<MonitorProperty> props) : this(monitorData, queries) => MonitorProperties = props;
        public int MonitorKey { get; }
        public string MonitorName { get; }
        public MonitorType MonitorType { get; }
        public MonitorLevel MonitorLevel { get; }
        public MonitorCategories MonitorCategory { get; }
        public List<Query> Queries { get; } = new List<Query>();
        public List<MonitorProperty> MonitorProperties { get; } = new List<MonitorProperty>();
    }
}

File: QIQO.Monitor.Client/ClientApp/src/app/models/monitor.ts
/* eslint-disable no-shadow */
/* eslint-disable @typescript-eslint/naming-convention */
import { Query } from './query';
import { ResultInstance } from './result-instance';

export enum MonitorLevel {
  Instance = 1,
  Database
}

export enum MonitorCategories {
  Version = 1,
  SQLServerHardware,
  DetectBlocking,
  OpenTranactions,
  WaitStatistics
}

export interface Monitor {
  monitorKey: number;
  monitorName: string;
  monitorType: number;
  monitorLevel: MonitorLevel;
  monitorCategory: MonitorCategories;
  queries: Query[];
  monitorProperties: MonitorProperty[];
  lastMonitorResult: ResultInstance;
}

export interface MonitorCategory {
  categoryKey: number;
  categoryName: string;
}

export interface MonitorProperty {
  propertyType: string;
  propertyDataType: string;
  propertyValue: string;
}
